tests/components/onvif/test_device_trigger.py

```diff
@@ -0,0 +1,124 @@
+    )
+    onvif_config_entry.add_to_hass(hass)
+    device_entry = device_reg.async_get_or_create(
+        config_entry_id=onvif_config_entry.entry_id,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, "12:34:56:AB:CD:EF")},
+    )
+
+    # Add another config entry for this device
+    unifi_config_entry = MockConfigEntry(
+        domain="unifi", unique_id="12:34:56:AB:CD:EF", data={}
+    )
+    unifi_config_entry.add_to_hass(hass)
+    device_reg.async_get_or_create(
+        config_entry_id=unifi_config_entry.entry_id,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, "12:34:56:AB:CD:EF")},
+    )
+
+    # Create device and mock event
+    device = ONVIFDevice(hass, onvif_config_entry)
+    device.events = EventManager(hass, None, onvif_config_entry.unique_id)
+    # pylint: disable=protected-access
+    device.events._events = {
+        f"{onvif_config_entry.unique_id}_tns1:RuleEngine/LineDetector/Crossed_0_0_0": Event(
+            f"{onvif_config_entry.unique_id}_tns1:RuleEngine/LineDetector/Crossed_0_0_0",
+            "0 Line Crossed",
+            "event",
+        )
+    }
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][onvif_config_entry.unique_id] = device
+
+    # Only expect triggers for ONVIF domain
+    expected_triggers = [
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "event",
+            "subtype": "0 Line Crossed",
+            "device_id": device_entry.id,
+            "unique_id": f"{onvif_config_entry.unique_id}_tns1:RuleEngine/LineDetector/Crossed_0_0_0",
+        },
+    ]
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    assert_lists_same(triggers, expected_triggers)
+
+
+async def test_if_fires_on_event(hass, calls):
+    """Test that onvif_event triggers firing."""
+    event_uid = "12:34:56:AB:CD:EF_tns1:RuleEngine/LineDetector/Crossed_0_0_0"
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": "",
+                        "type": "event",
+                        "subtype": "0 Line Crossed",
+                        "unique_id": event_uid,
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "some": (
+                                "{{ trigger.platform}} - "
+                                "{{ trigger.event.event_type}} - {{ trigger.event.data.unique_id}}"
+                            )
+                        },
+                    },
+                },
+            ]
+        },
+    )
+
+    # Fire event
+    hass.bus.fire(CONF_ONVIF_EVENT, {CONF_UNIQUE_ID: event_uid})
```

Comment:
Please use the async api.
```suggestion
    hass.bus.async_fire(CONF_ONVIF_EVENT, {CONF_UNIQUE_ID: event_uid})
```